bin/ch-run.c

```diff
@@ -184,6 +187,36 @@ void env_delta_append(struct env_delta **ds, enum env_action act, char *arg)
    (*ds)[i].arg = arg;
 }
 
+/* Parse and set input from --set-env */
+void env_parse(char *name, char *new_value, bool expand)
+{
+   char *token;
+   char *new_env = "";
+   const char s[1] = ":";
+   if (   strlen(new_value) >= 2
+       && (new_value)[0] == '\''
+       && (new_value)[strlen(new_value) - 1] == '\'') {
+      (new_value)[strlen(new_value) - 1] = 0;  // strip trailing quote
+      (new_value)++;                           // strip leading
+   }
+   token = strtok(new_value, s);
+   while( token != NULL ) {
+      if (token[0] == '$' && expand) {
+         token ++;
+         if(getenv(token) != NULL)
+            T_ (1 <= asprintf(&new_env, "%s:%s", new_env, getenv(token)));
+      }
+      else {
+         T_ (1 <= asprintf(&new_env, "%s:%s", new_env, token));
+      }
+      token = strtok(NULL, s);
+   }
+   if (new_env[0] == ':')
+      new_env++;
```

Comment:
This doesn't quite match the docs. We can change either but I think you are probably better off omitting the two lines and dealing with it elsewhere.